src/main/java/de/qabel/desktop/ui/AbstractController.java

```diff
@@ -150,4 +152,29 @@ public Contact gsonContactToContact(GsonContact gc, Identity i) throws URISyntax
 		}
 		return collection;
 	}
+	protected String calculateTimeString(DropMessage dropMessage) {
+
+		Long messageDate = dropMessage.getCreationDate().getTime();
+
+		java.util.Date date= new java.util.Date();
+		Timestamp currentTimestamp = new Timestamp(date.getTime());
+
+		long diff = currentTimestamp.getTime() - messageDate;
+
+		int minutes = (int) ((diff / (1000*60)));
+		int hours   = (int) ((diff / (1000*60*60)));
+
+		String text;
+		if (minutes < 10) {
+			text = "moments ago";
+		} else if (minutes < 60) {
```

Comment:
The output is untranslated and most branches of this methos seem untested.
Probably the use of a lib to handle the human time conversion is better (because it exists and is tested). Like http://www.ocpsoft.org/prettytime/ or https://github.com/mfornos/humanize ). Probably at least one of them should support translations...